### ISSUE TITLE
Domogik's process take a long time before stopping completely

### ISSUE THREAD

**Issue by vdomos** [domogik/domogik#233]
To answer the question addressed here: https://github.com/vdomos/domogik-plugin-script/issues/4

When I run a **/etc/init.d/domogik stop**, it's very long to stop all process, especially if they run for a while.

That's why, I don't use **/etc/init.d/domogik restart** to restart Domogik.

Here a example (~1mn 30s to stop all process):

```
root@hermes:~# /etc/init.d/domogik stop ; while true; do echo ;date;ps ax|grep -Ei '[d]omo|[d]mg'; sleep 5;done
* Stopping Plugins... Done.
* Stopping Manager... Done.
* Stopping XPL... Done.
* Stopping 0MQ broker... Done.
* Stopping 0MQ forwarder... Done.

lundi 15 février 2016, 15:03:29 (UTC+0100)
  999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:03:34 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:03:39 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:03:44 (UTC+0100)                                                                                                                                                                        
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:03:49 (UTC+0100)                                                                                                                                                                        
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:03:54 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:03:59 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:04 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:09 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:15 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:20 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:25 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
25302 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_vdevice/bin/vdevice.py
25336 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_k8056/bin/k8056.py
25377 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rfxbnz/bin/rfxbnz.py
25418 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_script/bin/script.py
25457 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_weather/bin/weather.py
25498 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_rainhour/bin/rainhour.py
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:30 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
1069 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_onewired/bin/onewired.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:35 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:40 (UTC+0100)
999 ?        D      0:00 /usr/bin/python /var/lib/domogik//domogik_packages/plugin_mqtt/bin/mqtt.py
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
26073 ?        Sl    24:13 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:45 (UTC+0100)
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
26073 ?        Sl    24:14 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:50 (UTC+0100)
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25098 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/xplgw.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25160 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/scenario.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
26073 ?        Sl    24:14 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:04:55 (UTC+0100)
25053 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/dbmgr.pyc
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
25196 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/butler.pyc
26073 ?        Sl    24:14 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:05:00 (UTC+0100)
25126 ?        D      0:00 /usr/bin/python /opt/dmg/domogik/src/domogik/bin/admin.pyc
26073 ?        Sl    24:14 /usr/bin/python /opt/dmg/domoweb/Domoweb.py

lundi 15 février 2016, 15:05:05 (UTC+0100)
26073 ?        Sl    24:14 /usr/bin/python /opt/dmg/domoweb/Domoweb.py
```

**Comment by fritz-smh** [comment]
That is quite strange, I don't have this issue on both my dev laptop and
nas server...
Can you do this :

* kill all
* clean the logs (or backup them if you want to keep them)
* start domogik
* wait 5 minutes
* launch the 'uptime' command (to get load average before shutting)
* launch : "time /etc/init.d/domogik stop ; uptime"   and give us the full
output of the command
* attach all the log files to this issue

Thanks,
Fritz

**Comment by vdomos** [comment]
I do not think it will be the same thing to stop Domogik after 5 minutes of operation.
The stop is quick in this case.
it will take time only if Domogik running for a while. In this test there was> 310000 open files given by lsof before stopping Domogik and only 20386 after.
When I haved > 800000, the stop was even longer.

This is perhaps the closure of all open files that delay the stop of Domogik ?

I can test after reboot the VM to clean all, running domogik a day or more , and do a stop again no ?

**Comment by fritz-smh** [comment]
what are the plugins installed on your system ?

ok for the test

**Comment by vdomos** [comment]
plugin_callerid     1 device
    plugin_k8056        1 device
    plugin_mqtt         3 devices
    plugin_onewired     12 devices                                                                                                                              
    plugin_rfxbnz       14 devices  (only devices, no plugin)                                                                                                                                       
    plugin_script       13 devices                                                                                                                                  
    plugin_vdevice      8 devices                                                                                                                                
    plugin_weather      2 devices

**Comment by tikismoke** [comment]
script
wol
notify
nabaztag
onewired
ozwave
karotz
generic
plcbus
ping
diskfree
mirror
weather
teleinfo
vdevice

**Comment by vdomos** [comment]
Too strong for me Tikismoke :)

**Comment by fritz-smh** [comment]
tiki : you also have the issue ?

**Comment by tikismoke** [comment]
Yes but only with script. I always use restart.

**Comment by vdomos** [comment]
tiki : can you do the same test ?
Let Domogik run a day or more, do a domogik stop and see if the domogik process take time to stop.

**Comment by fritz-smh** [comment]
before shutting, can you also do a : ps -efL | grep a_plugin
with each of your plugin name as 'a_plugin'

The L option will display all threads off the plugins. Too much threads in one plugin could explain this ;)

**Comment by Cereal2nd** [comment]
hmm, after some debugging i noticed that manager is not stopping the plugins, the init script is killing them eventually.

Thsi si not correct and should be fixed

**Comment by fritz-smh** [comment]
Do we need to fix this one for 0.5 release or could we postpone it to 0.6 ?

**Comment by Cereal2nd** [comment]
i think this is important to fix this in 0.5

the manager should stop all plugins it started when its stopped
so we should catch the kill signal and then stop the plugins

**Comment by Cereal2nd** [comment]
i think its related to #79

**Comment by Cereal2nd** [comment]
it should be solved in the nowatcher branch, can you test?

**Comment by vdomos** [comment]
it's good for me, the plugins are stopped quickly.

**Comment by Cereal2nd** [comment]
Ok, this will be included in 0.6 then

**Comment by Cereal2nd** [comment]
noWatcher is merged in, so closing the ticket
